### PR TITLE
Add general process for printing HU labels 

### PR DESCRIPTION
**Pull request by cristinamghita**

### CODE REVIEW

File: backend/de.metas.handlingunits.base/src/main/java/de/metas/handlingunits/process/M_HU_Report_Print_Labels.java [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+	private final IQueryBL queryBL = Services.get(IQueryBL.class);
+
+
+	@Param(mandatory = true, parameterName = "AD_Process_ID")
+	private int p_AD_Process_ID;
+
+	@Override
+	public ProcessPreconditionsResolution checkPreconditionsApplicable(final @NonNull IProcessPreconditionsContext context)
+	{
+		if (context.isMoreThanOneSelected())
```
Comment (adi-stefan) [inline backend/de.metas.handlingunits.base/src/main/java/de/metas/handlingunits/process/M_HU_Report_Print_Labels.java:70]: should it work if none are selected?

File: backend/de.metas.ui.web.base/src/main/java/de/metas/ui/web/process/adprocess/ADProcessDescriptorsFactory.java [hunk 2]
```diff
@@ -247,6 +244,26 @@ private ProcessDescriptor retrieveProcessDescriptor(final ProcessId processId)
 				.build();
 	}
 
+	public void addProcessParameters(final ProcessId processId, final DocumentEntityDescriptor.Builder parametersDescriptorBuilder)
```
Comment (adi-stefan) [inline backend/de.metas.ui.web.base/src/main/java/de/metas/ui/web/process/adprocess/ADProcessDescriptorsFactory.java:247]:
```suggestion
	public void addProcessParameters(@NonNull final ProcessId processId, @NonNull final DocumentEntityDescriptor.Builder parametersDescriptorBuilder)
```

File: backend/de.metas.ui.web.base/src/main/java/de/metas/ui/web/process/adprocess/ADProcessDescriptorsFactory.java [hunk 3]
```diff
@@ -247,6 +244,26 @@ private ProcessDescriptor retrieveProcessDescriptor(final ProcessId processId)
...
+		if (adProcess == null)
+		{
+			throw new EntityNotFoundException("@NotFound@ @AD_Process_ID@ (" + processId + ")");
+		}
+
+		final WebuiProcessClassInfo webuiProcesClassInfo = WebuiProcessClassInfo.of(adProcess.getClassname());
+		addProcessParameters(processId, webuiProcesClassInfo, parametersDescriptorBuilder);
+	}
+
+	private void addProcessParameters(final ProcessId processId, final WebuiProcessClassInfo webuiProcesClassInfo, final DocumentEntityDescriptor.Builder parametersDescriptorBuilder)
```
Comment (adi-stefan) [inline backend/de.metas.ui.web.base/src/main/java/de/metas/ui/web/process/adprocess/ADProcessDescriptorsFactory.java:259]:
```suggestion
	private void addProcessParameters(@NonNull final ProcessId processId, @NonNull final WebuiProcessClassInfo webuiProcesClassInfo, @NonNull final DocumentEntityDescriptor.Builder parametersDescriptorBuilder)
```

File: backend/de.metas.ui.web.base/src/main/java/de/metas/ui/web/process/adprocess/ADProcessInstancesRepository.java [hunk 4]
```diff
@@ -467,4 +467,10 @@ public <R> R forProcessInstanceWritable(final DocumentId pinstanceId, final IDoc
 			}
 		}
 	}
+
+
+	public void addProcessParameters(final ProcessId processId, final DocumentEntityDescriptor.Builder parametersDescriptorBuilder)
```
Comment (adi-stefan) [inline backend/de.metas.ui.web.base/src/main/java/de/metas/ui/web/process/adprocess/ADProcessInstancesRepository.java:472]:
```suggestion
	public void addProcessParameters(@NonNull final ProcessId processId, @NonNull final DocumentEntityDescriptor.Builder parametersDescriptorBuilder)
```

File: backend/de.metas.ui.web.base/src/main/java/de/metas/ui/web/process/view/ViewActionDescriptor.java [hunk 5]
```diff
@@ -102,6 +99,14 @@ public String getActionId()
...
 	}
 
+	public void addParametersDescriptor(final DocumentEntityDescriptor.Builder parametersDescriptor)
+	{
+		viewActionParamDescriptors.stream()
+				.filter(ViewActionParamDescriptor::isUserParameter)
+				.map(ViewActionParamDescriptor::createParameterFieldDescriptor)
+				.forEach(parametersDescriptor::addField);
+	}
+
```
Comment (adi-stefan) [inline backend/de.metas.ui.web.base/src/main/java/de/metas/ui/web/process/view/ViewActionDescriptor.java:109]: I think this is no longer used, right? aka, can be reverted.

**Review by adi-stefan — CHANGES_REQUESTED** [review]
Please see minor comments above.

File: backend/de.metas.ui.web.base/src/main/java/de/metas/ui/web/process/view/ViewActionDescriptor.java (hunk 5 above)
Comment (cristinamghita) [reply backend/de.metas.ui.web.base/src/main/java/de/metas/ui/web/process/view/ViewActionDescriptor.java:109]: Still used

File: backend/de.metas.handlingunits.base/src/main/java/de/metas/handlingunits/process/M_HU_Report_Print_Labels.java (hunk 1 above)
Comment (cristinamghita) [reply backend/de.metas.handlingunits.base/src/main/java/de/metas/handlingunits/process/M_HU_Report_Print_Labels.java:70]: No, and the process is not there if no line is selected.

**Review by adi-stefan — COMMENTED** [review]
Understood that no budget for the params thing, so LGTM.

File: backend/de.metas.handlingunits.base/src/main/java/de/metas/handlingunits/process/M_HU_Report_Print_Labels.java [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+
+		return ProcessPreconditionsResolution.accept();
+	}
+
+	@Override
+	@RunOutOfTrx
+	protected String doIt()
+	{
+
+		final HuId huId = queryBL.createQueryBuilder(I_M_HU.class)
```
Comment (teosarca) [inline backend/de.metas.handlingunits.base/src/main/java/de/metas/handlingunits/process/M_HU_Report_Print_Labels.java:83]: @cristinamghita  next time pls avoid using queryBL from processes. That code chunk should go to HandlingUnitsDAO

**Pull request merged**